### PR TITLE
Implementation fromNow

### PR DESCRIPTION
**Pull request by naqvisat**
Implementation fromNow for DateTime, Date and Time

### CODE REVIEW

File: shared/src/main/scala/pl/metastack/metatime/Constants.scala [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package pl.metastack.metatime
+
+object Constants {
+    final val YEAR_INDEX = 1
```
Comment (tindzk) [inline shared/src/main/scala/pl/metastack/metatime/Constants.scala]: Can you write `YearIndex`, `MonthIndex`, etc.?

The indentation is off. It should be only two characters.

File: shared/src/main/scala/pl/metastack/metatime/Date.scala [hunk 2]
```diff
@@ -106,8 +106,9 @@ object Date {
     }
   }
 
-  def findMonth(days: Int, year: Int): Int =
-    accumulate(accuDays, year).filter(x => x <= days).length + 1
+  def findMonth(days: Int, year: Int): Int = {
+    accumulate(accuDays, year).filter(x => x <= math.abs(days)).length + 1
```
Comment (tindzk) [inline shared/src/main/scala/pl/metastack/metatime/Date.scala]: You should not pass a negative value in the first place when calling `findMonth()`. You could add `require(days >= 0 && year >= 0)` to verify the value ranges.

File: shared/src/main/scala/pl/metastack/metatime/Offset.scala [hunk 3]
```diff
@@ -15,6 +17,34 @@ object Formatter {
   def branch(ranges: Seq[(Range, Any)], fallback: Any): Formatter = ???
 }
 
-trait Offset[T] {
-  def format: String = ???
+trait Offset[+T] {
```
Comment (tindzk) [inline shared/src/main/scala/pl/metastack/metatime/Offset.scala]: Can you add a type constraint (`trait Offset[+T <: Component]`)? Then the cast in line 23 should become redundant.

File: shared/src/main/scala/pl/metastack/metatime/Time.scala [hunk 4]
```diff
@@ -20,31 +22,167 @@ trait Component extends Ordered[Component] {
...
-    if(unixDiff <= 86400000) {
-      val timeObj = Time(unixDiff)
-      val timeNow = Time(biggerTime)
-      if((timeObj.h == timeNow.h) && (timeObj.m == timeNow.m)){
-        Second(timeObj.s)
-      }
-      else if(timeObj.h == timeNow.h) {
-        Minute(timeNow.m - timeObj.m)
-      }
+  def fromNow(): Offset[_] = {
```
Comment (tindzk) [inline shared/src/main/scala/pl/metastack/metatime/Time.scala]: I believe you can shorten this function to:

``` scala
def fromNow(): Offset[_] = Offset(calcOffset())
```

File: shared/src/main/scala/pl/metastack/metatime/Offset.scala (hunk 3 above)
Comment (naqvisat) [inline shared/src/main/scala/pl/metastack/metatime/Offset.scala]: Actually I tried that but I got this error:

`\Offset.scala:47: type arguments [Component] do not conform to trait Offset's type parameter bounds [T <: pl.metastack.metatime.Component]
[error]   def apply[Component](value: Component): Offset[Component] = new Offset[Component] {
[error]   def apply[Component](value: Component): Offset[Component] = new Offset[Component] {`

that's why I didn't include T <: Compoent, any idea how to solve it further?

Comment (tindzk) [inline shared/src/main/scala/pl/metastack/metatime/Offset.scala]: The problem is that you define a generic type parameter `Component` in line 47 which is not a sub-type of the trait `Component`. The generic has the same name, but doesn't actually point to the `Component` trait as you may expect.

This works:

``` scala
def apply(value: Component): Offset[Component] = new Offset[Component] { ... }
```

Or even better:

``` scala
def apply[T <: Component](value: T): Offset[T] = new Offset[T] { ... }
```

Comment (naqvisat) [inline shared/src/main/scala/pl/metastack/metatime/Offset.scala]: cool, works.

**Pull request merged**